.github/PULL_REQUEST_TEMPLATE/pull_request.md

```diff
@@ -0,0 +1,24 @@
+---
+name: Pull request
+about: Create a pull request for an issue in Altinn Studio
+---
+
+Thank you for contributing to Altinn:) We know this isn't the fun part, 
+but please put the same effort into the pull request as you did into the code and it should soon find it's way to master.
+
+# {PR title}
+
+Summary of the changes (max 80 characters)
+
+## Description
+
+{Detail}
+
+Fixes #{issue number} (in this specific format)
+
+## Verification
+- [ ] Your code builds clean without any errors or warnings
+- [ ] Manual testing done (required)
+- [ ] Autmated test added (not strictly required, but you should have a very good reason not to)
```

Comment:
I dont see why we should not require automated tests to be added - IMO we should be stricter on this moving forward. We should be able to rely on all code in our codebase, and not be scared to make changes to areas because they dont have test coverage.

Reply:
But we do have some cleanup to do in the codebase first (at least from a frontend perspective - its not straight forward to run lint/tests), to make this process easier. Perhaps it can be encouraged but not enforced for now :)

Reply:
100% agree, what I had in mind here was changes either are covered by automatic tests already or changes in for example build scripts, local setup, doc etc. that don't require automated tests. But we could change the wording and leave it up to people to think about those cases in stead of leaving it as an option (which it really isn't).

Reply:
A good reason not to require new automated tests is that it is another barrier for people that wants to contribute, and might reduce the number of people that submits their first PR. Everybody starts with a few poor quality pull requests before they get to know the system. Internal team members gets a few days of mentoring and introduction that external contributors does not get. If you want the contributor base to grow, you need to have lots of people have a nice experience so they get a feeling that they can improve the system for the better.

Reply:
I have not yet found your docs on your different test suites, how to install dependencies and run them, and where it is relevant to add tests for different changes, so for me it's definitely a barrier. Maybe I'm just not good enough to be valuable.

Reply:
Agree on the docs and that we definitely can get better in making it easier to add tests. Disagree on you not being good enough to be valuable:)

But maybe our checklist is a bit of wishful thinking on where we would like to be and not entirely in sync with where we're currently at with regards to both the codebase and the docs. A goal for us with making this open source is absolutely to make it easy with external contributions at the same time strike a balance with a reasonable checklist for those contributions.